altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -0,0 +1,426 @@
+// Daniel Seitz and RJ Russell
+
+use super::super::Register;
+use super::defs::*;
+
+/// Three USART control registers.
+#[derive(Copy, Clone)]
+pub struct UsartCR {
+    cr1: CR1,
+    cr2: CR2,
+    cr3: CR3,
+}
+
+// TODO Need to implement a clear mask for each register to ensure that
+// all register bits are set to zero before re-initializing register to
+// necessary values for a specific usart configuration.
+impl UsartCR {
+    pub fn new(base_addr: *const u32) -> Self {
+        UsartCR {
+            cr1: CR1::new(base_addr),
+            cr2: CR2::new(base_addr),
+            cr3: CR3::new(base_addr),
+        }
+    }
+
+    pub fn enable_usart(&self) {
+        self.cr1.enable_usart(true);
+    }
+
+    pub fn disable_usart(&self) {
+        self.cr1.enable_usart(false);
+    }
+
+    pub fn is_usart_enabled(&self) -> bool {
+        self.cr1.is_usart_enabled()
+    }
+
+    pub fn set_word_length(&self, length: WordLength) {
+        self.cr1.set_word_length(length);
+    }
+
+    pub fn set_mode(&self, mode: Mode) {
+        self.cr1.set_mode(mode);
+    }
+
+    pub fn set_parity(&self, parity: Parity) {
+        self.cr1.set_parity(parity);
+    }
+
+    pub fn set_stop_bits(&self, length: Stoplength) {
+        self.cr2.set_stop_bits(length);
+    }
+
+    pub fn enable_over8(&self) {
+        self.cr1.set_over8(true);
+    }
+
+    pub fn disable_over8(&self) {
+        self.cr1.set_over8(false);
+    }
+
+    pub fn get_over8(&self) -> bool {
+        self.cr1.get_over8()
+    }
+
+    pub fn set_hardware_flow_control(&self, hfc: HardwareFlowControl) {
+        self.cr3.set_hardware_flow_control(hfc);
+    }
+}
+
+// ------------------------------------
+/// CR1
+// ------------------------------------
+
+/// Word length can be 7, 8, or 9 bits.
+#[derive(Copy, Clone)]
+pub enum WordLength {
```

Comment:
Yeah, I can see that, I was thinking more from a usage point of view. People trying to use the module probably want to see the definitions, so I was just thinking it'd be easiest to find them when they're at the top. Since we're using Rust docs to generate documentation this shouldn't be an issue though.